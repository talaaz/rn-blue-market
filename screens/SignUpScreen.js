/*
 Acknowlegdement:
 For design: https://www.chatkitty.com/blog/posts/building-a-chat-app-with-react-native-and-firebase-part-1/
 For uploadImage: https://github.com/SimCoderYoutube/InstagramClone/blob/master/frontend/components/main/Save.js

C:/Users/Valde/Documents/LAPTOP-PLFGULU9/Documents/It-elektronik/5. Semester/Selvstudie i Mobilapplikationsudvikling/BlueMarket/rn-blue-market/assets/default_profile_pic.jpg
*/

import React, { useEffect, useState } from "react";
import { StyleSheet, View, Image } from "react-native";
import { IconButton, Title, Text, Avatar, Snackbar } from "react-native-paper";
import * as Permissions from "expo-permissions";
import * as ImagePicker from "expo-image-picker";

import FormButton from "../components/FormButton";
import FormInput from "../components/FormInput";
import Loading from "../components/Loading";

import { firebase } from "../firebase";

export default function SignupScreen({ navigation }) {
  const [
    camPermission,
    askForCameraPermission,
    getCameraPermission,
  ] = Permissions.usePermissions(Permissions.CAMERA, { ask: true });

  const [
    storagePermission,
    askForStoragePermission,
    getStoragePermission,
  ] = Permissions.usePermissions(Permissions.MEDIA_LIBRARY, { ask: true });

  const [credential, setCredential] = useState(null);
  const [localProfileURL, setLocalProfileURL] = useState("");
  const [cloudProfileURL, setCloudProfileURL] = useState();

  const [defaultPic, setDefaultPic] = useState(true);
  const [profilePic, setProfilePic] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [loading, setLoading] = useState(false);

  const [visible, setVisible] = useState(false);
  const [errorCode, setErrorCode] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    (async () => {
      // Fetch cloud based profilePic.
      if (defaultPic) {
        await firebase
          .storage()
          .ref()
          .child("images/profilepic.jpg")
          .getDownloadURL()
          .then((url) => {
            setProfilePic(url);
          });
      }
    })();
  });

  const onDismissSnackBar = () => setVisible(false);

  const takePictureWithCamera = async () => {
    console.log("Opening Camera");
    if (!camPermission || camPermission.status !== "granted") {
      setVisible(true);
      setErrorMessage("Permission for accessing phone camera wasn't granted");
      return;
    }
    //Permission granted open camera.
  };

  const fetchPictureFromGallery = async () => {
    // Check media storage permissions. Send snackbar error if not given.
    console.log("Opening Gallery");
    if (!storagePermission || storagePermission.status !== "granted") {
      setVisible(true);
      setErrorMessage("Permission for accessing phone storage wasn't granted");
      return;
    }
    //Permission granted open image-picker.
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });
    if (!result.cancelled) {
      setDefaultPic(false);
      setProfilePic(result.uri);
    }
  };

  const uploadImage = async () => {
    const childPath = `users/${
      firebase.auth().currentUser.uid
    }/profile/profilepic.jpg`;
    console.log(childPath);

    const response = await fetch(profilePic);
    const blob = await response.blob();

    const task = firebase.storage().ref().child(childPath).put(blob);

    const taskProgress = (snapshot) => {
      console.log(`transferred: ${snapshot.bytesTransferred}`);
    };

    const taskCompleted = () => {
      task.snapshot.ref.getDownloadURL().then((url) => {
        setProfilePic(url);
        console.log(url);
      });
    };

    const taskError = (snapshot) => {
      setVisible(true);
      setErrorCode(snapshot.error.code);
      setErrorMessage(snapshot.error.message);
      console.log(snapshot);
    };

    task.on("state_changed", taskProgress, taskError, taskCompleted);
  };

  const register = async (firstName, lastName, email, password) => {
    setLoading(true);
    try {
      await firebase
        .auth()
        .createUserWithEmailAndPassword(email, password)
        .then((credential) => {
          uploadImage();
          credential.user
            .updateProfile({
              displayName: firstName + " " + lastName,
              photoURL: profilePic,
            })
            .then(async () => {
              // Go back to log in screen after succesfully created an user.

              navigation.goBack();
            });
        });
    } catch (e) {
      setVisible(true);
      setErrorCode(e.code);
      setErrorMessage(e.message);
      console.log(e);
    }

    setLoading(false);
  };

  if (loading) {
    return <Loading />;
  }

  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Avatar.Image size={100} source={{ uri: profilePic }}></Avatar.Image>
      </View>
      <View style={styles.linkContainer}>
        <Text onPress={() => takePictureWithCamera()} style={styles.linkText}>
          {"Take picture"}
        </Text>
        <Text onPress={() => fetchPictureFromGallery()} style={styles.linkText}>
          {"Get picture from Gallery"}
        </Text>
        <Text onPress={() => setDefaultPic(true)} style={styles.linkText}>
          {"Set default picture"}
        </Text>
      </View>
      <View style={styles.formContainer}>
        <Title style={styles.titleText}>Let's get started!</Title>
        <FormInput
          labelName="Firstname"
          value={firstName}
          autoCapitalize="none"
          onChangeText={(userFirstName) => setFirstName(userFirstName)}
        />
        <FormInput
          labelName="Lastname"
          value={lastName}
          autoCapitalize="none"
          onChangeText={(userLastName) => setLastName(userLastName)}
        />
        <FormInput
          labelName="Email"
          value={email}
          autoCapitalize="none"
          onChangeText={(userEmail) => setEmail(userEmail)}
        />
        <FormInput
          labelName="Password"
          value={password}
          secureTextEntry={true}
          onChangeText={(userPassword) => setPassword(userPassword)}
        />
        <FormButton
          title="Signup"
          modeValue="contained"
          labelStyle={styles.loginButtonLabel}
          onPress={() => register(firstName, lastName, email, password)}
        />
        <IconButton
          icon="keyboard-backspace"
          size={30}
          style={styles.navButton}
          color="#5b3a70"
          onPress={() => navigation.goBack()}
        />
      </View>
      <Snackbar
        visible={visible}
        onDismiss={onDismissSnackBar}
        action={{
          label: "Dismiss",
          onPress: () => {
            // Do something
          },
        }}
      >
        {errorMessage}
      </Snackbar>
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    backgroundColor: "#f5f5f5",
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    alignItems: "center",
  },
  linkContainer: {
    justifyContent: "space-around",
    flexDirection: "column",
    alignItems: "center",
  },

  formContainer: {
    alignItems: "center",
  },

  image: {
    flex: 1,
    height: undefined,
    width: undefined,
  },

  linkText: {
    fontSize: 14,
    color: "blue",
    textDecorationLine: "underline",
  },

  titleText: {
    fontSize: 24,
    marginBottom: 10,
  },
  loginButtonLabel: {
    fontSize: 22,
  },
  navButtonText: {
    fontSize: 18,
  },
  navButton: {
    marginTop: 10,
  },
});
