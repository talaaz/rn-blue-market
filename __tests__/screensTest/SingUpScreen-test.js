import React from "react";
import renderer from "react-test-renderer";
import SignUpScreen from "../../screens/SignUpScreen";

jest.useFakeTimers();
describe("<SignUpScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SignUpScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
