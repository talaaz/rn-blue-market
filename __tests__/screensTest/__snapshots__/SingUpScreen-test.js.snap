// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "color": "blue",
            "fontSize": 14,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Take picture
    </Text>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "color": "blue",
            "fontSize": 14,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Get picture from Gallery
    </Text>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Object {
            "color": "blue",
            "fontSize": 14,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      Set default picture
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 24,
                "marginBottom": 10,
              },
            ],
          ],
        ]
      }
    >
      Let's get started!
    </Text>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": 500,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#643ca4",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 88.93333333333334,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 53,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Firstname
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Firstname
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Firstname
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="#643ca4"
            rejectResponderTermination={true}
            selectionColor="#643ca4"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "zIndex": 1,
                },
                Object {
                  "height": 88.93333333333334,
                },
                Object {
                  "paddingBottom": 36.46666666666667,
                  "paddingTop": 36.46666666666667,
                },
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": 500,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#643ca4",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 88.93333333333334,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 53,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Lastname
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Lastname
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Lastname
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="#643ca4"
            rejectResponderTermination={true}
            selectionColor="#643ca4"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "zIndex": 1,
                },
                Object {
                  "height": 88.93333333333334,
                },
                Object {
                  "paddingBottom": 36.46666666666667,
                  "paddingTop": 36.46666666666667,
                },
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": 500,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#643ca4",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 88.93333333333334,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 53,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Email
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Email
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Email
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="#643ca4"
            rejectResponderTermination={true}
            selectionColor="#643ca4"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "zIndex": 1,
                },
                Object {
                  "height": 88.93333333333334,
                },
                Object {
                  "paddingBottom": 36.46666666666667,
                  "paddingTop": 36.46666666666667,
                },
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": 500,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "#643ca4",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 88.93333333333334,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 53,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Password
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Password
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 52,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Password
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="#643ca4"
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="#643ca4"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "zIndex": 1,
                },
                Object {
                  "height": 88.93333333333334,
                },
                Object {
                  "paddingBottom": 36.46666666666667,
                  "paddingTop": 36.46666666666667,
                },
                Object {
                  "color": "#643ca4",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlign": "left",
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#643ca4",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginTop": 10,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "height": 88.93333333333334,
                "width": 375,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontSize": 22,
                  },
                ],
              ]
            }
          >
            Signup
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 22.5,
              "height": 45,
              "width": 45,
            },
            undefined,
            Object {
              "marginTop": 10,
            },
          ],
        ]
      }
    >
      <View>
        <Text />
      </View>
    </View>
  </View>
</View>
`;
